components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -12853,53 +12857,105 @@ public void addAPIProduct(APIProduct apiproduct) throws APIManagementException {
             prepStmtAddAPIProduct.setString(7, apiproduct.getSubscriptionAvailability());
             prepStmtAddAPIProduct.setString(8, apiproduct.getUuid());
             prepStmtAddAPIProduct.execute();
-            connection.commit(); //TODO check this///////
-            
-            //Get the product id related to the product
-            //TODO check this
-            //TODO move to constant
-            String queryGetAPIProduct = "SELECT API_PRODUCT_ID FROM AM_API_PRODUCT WHERE UUID = ?";
-            
-            prepStmtGetAPIProduct = connection.prepareStatement(queryGetAPIProduct);
-            prepStmtGetAPIProduct.setString(1, apiproduct.getUuid());
-            rs = prepStmtGetAPIProduct.executeQuery();
+
+            rs = prepStmtAddAPIProduct.getGeneratedKeys();
+
             if (rs.next()) {
-                productId = rs.getInt("API_PRODUCT_ID");
+                productId = rs.getInt(1);
```

Comment:
shall we use the column instead of the column id. column name would improve readability/maintainability

Reply:
Here I am retrieving the autogenerated API product id key. Autogenerated key column value has to be accessed by column index.